ds3-contract-comparator-print/src/main/java/com/spectralogic/ds3contractcomparator/print/htmlprinter/HtmlReportPrinter.java

```diff
@@ -0,0 +1,210 @@
+import com.spectralogic.ds3contractcomparator.models.type.AbstractDs3TypeDiff;
+import com.spectralogic.ds3contractcomparator.models.type.AddedDs3TypeDiff;
+import com.spectralogic.ds3contractcomparator.models.type.DeletedDs3TypeDiff;
+import com.spectralogic.ds3contractcomparator.models.type.Ds3TypeDiff;
+import com.spectralogic.ds3contractcomparator.models.type.ModifiedDs3TypeDiff;
+import com.spectralogic.ds3contractcomparator.print.Ds3SpecDiffPrinter;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.generators.HtmlRequestTableGenerator;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.generators.HtmlTypeTableGenerator;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.HtmlReport;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.body.Section;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.body.Table;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.index.IndexEntry;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.index.IndexSection;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateExceptionHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import static com.spectralogic.ds3autogen.utils.NormalizingContractNamesUtil.removePath;
+import static com.spectralogic.ds3contractcomparator.print.utils.HtmlPrinterUtils.toRequestAnchor;
+import static com.spectralogic.ds3contractcomparator.print.utils.HtmlPrinterUtils.toRequestTitle;
+
+/**
+ * Generates and prints the HTML report highlighting the differences between
+ * two {@link Ds3ApiSpec}.
+ */
+public class HtmlReportPrinter implements Ds3SpecDiffPrinter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HtmlReportPrinter.class);
+
+    private final Configuration config = new Configuration(Configuration.VERSION_2_3_23);
+    private final Writer writer;
+    private final String oldContractName;
+    private final String newContractName;
+
+    public HtmlReportPrinter(final Writer writer, final String oldContractName, final String newContractName) {
+        this.writer = writer;
+        this.oldContractName = oldContractName;
+        this.newContractName = newContractName;
+
+        config.setDefaultEncoding("UTF-8");
+        config.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
+        config.setClassForTemplateLoading(HtmlReportPrinter.class, "/tmpls");
+    }
+
+    @Override
+    public void print(final Ds3ApiSpecDiff specDiff) {
+        try {
+            generateReport(specDiff, oldContractName, newContractName, config, writer);
+        } catch (final Exception e) {
+            LOG.warn("Unable to print report: {}", e.getMessage());
```

Comment:
Make sure that the exception is captured in the log statement so that it looks like this:

```java

LOG.error("Unable to print report: " + e.getMessage(), e);
```